Parsimony.Tests/Builder/TestPropertySelector.cs

```diff
@@ -0,0 +1,64 @@
+﻿using Parsimony.ParserBuilder;
+using System;
+using System.Linq.Expressions;
+using Xunit;
+
+namespace Parsimony.Tests.Builder
+{
+    public class TestPropertySelector
+    {
+#nullable disable //Testing construction
+        [Fact]
+        public void Construct_SelectorNull_Throws()
+        {
+            Assert.Throws<ArgumentNullException>(() => new PropertySelector<TestDummy, bool>(null));
+        }
+#nullable enable
+
+        [Fact]
+        public void Construct_ExpressionIsNotMemberExpression_Throws()
+        {
+            Assert.Throws<ArgumentException>(() => Construct(x => x.GetMeADoot()));
+        }
+
+        [Fact]
+        public void Construct_ExpressionIsNotMemberAccessNodeType_Throws()
+        {
+            Assert.Throws<ArgumentException>(() => Construct(x => (int)x.IntProp));
```

Comment:
What's the difference between this and the one above? I would think that the cast makes this as much not a member expression as a literal, function call, etc. ?

Reply:
I have no idea. I'll remove this

Reply:
I had a separate check for the node type before. I can't remember why I had this. Probably vestigial. Will remove